### PR TITLE
Fix controllerregistrations permission issue

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix user

```

### CODE REVIEW

File: backend/lib/services/controllerregistrations.js [hunk 1]
```diff
@@ -6,23 +6,36 @@
...
 
 exports.listExtensions = async function ({ user }) {
-  const allowed = await authorization.canListControllerRegistrations(user)
-  if (!allowed) {
-    throw new Forbidden('You are not allowed to list controllerregistrations')
-  }
-
   const controllerregistrations = getControllerRegistrations()
-  return _.map(controllerregistrations, ({ metadata, spec }) => {
+  // required resoure kinds are essential for the frontend and need to be returned even if the user has not the right to read controllerregistrations
```
Comment (petersutter) [inline backend/lib/services/controllerregistrations.js:15]:
```suggestion
  // required resoure kinds are essential for the frontend and need to be returned even if the user has not the permission to read controllerregistrations
```

Comment (holgerkoser) [reply backend/lib/services/controllerregistrations.js:15]:
```js

const REQUIRED_RESOURCE_KINDS = ['Network', 'DNSProvider']

exports.listExtensions = async function ({ user }) {
  const controllerregistrations = getControllerRegistrations()
  const allowed = await authorization.canListControllerRegistrations(user)

  const extensions = []
  for (const { metadata, spec = {} } of controllerregistrations) {
    const name = metadata.name
    if (allowed) {
      const version = _.get(spec, 'deployment.providerConfig.values.image.tag')
      const resources = spec.resources
      extensions.push({ name, version, resources })
    } else {
      // required resoure kinds are essential for the frontend and need to be returned even if the user has not the permission to read controllerregistrations
      const resources = _.filter(spec.resources, ({ kind }) => REQUIRED_RESOURCE_KINDS.includes(kind))
      if (!_.isEmpty(resources)) {
        extensions.push({ name, resources })
      }
    }
  }
  return extensions
}
```

Comment (grolu) [reply backend/lib/services/controllerregistrations.js]: done

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Review by petersutter — APPROVED** [review]
/lgtm

**Pull request merged**